File: 215. Kth Largest Element in an Array/findKthLargest.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
const findKthLargest = (nums, k) => {
  const sorted = nums.sort((a, b) => b - a);
  return sorted[k - 1];
};

const nums = [3, 2, 1, 5, 6, 4];
const k = 2;
console.log('findKthLargest: ', findKthLargest(nums, k)); // 5

File: 215. Kth Largest Element in an Array/findKthLargest.py
import heapq
from typing import List

class Solution:
  def findKthLargest(self, nums: List[int], k: int) -> int:
    heap = []
    for num in nums:
      heapq.heappush(heap, num)
      if len(heap) > k:
        heapq.heappop(heap)

    return heap[0]
  

nums = [3,2,1,5,6,4]
k = 2
solution = Solution()
print('findKthLargest: ', solution.findKthLargest(nums, k)) # 5
